frontend/src/bundles/common/components/sidebar/hooks/use-collapse.hook.ts

```diff
@@ -0,0 +1,58 @@
+import { actions as authActions } from '~/bundles/auth/store/auth.js';
+import {
+    useAppDispatch,
+    useAppSelector,
+    useCallback,
+    useEffect,
+    useLayoutEffect,
+} from '~/bundles/common/hooks/hooks.js';
+import { storage, StorageKey } from '~/framework/storage/storage.js';
+
+import { MD } from '../constants/constants.js';
+import { getFlag } from '../helpers/helpers.js';
+
+const useCollapse = (): {
+    isCollapsed: boolean;
+    setToggle: () => void;
+} => {
+    const dispatch = useAppDispatch();
+    const isCollapsed = useAppSelector(({ auth }) => auth.isSidebarCollapsed);
+
+    const setToggle = (): void => {
+        dispatch(authActions.toggleSidebar(!isCollapsed));
+        void storage.set(
+            StorageKey.IS_SIDEBAR_COLLAPSED,
+            JSON.stringify(!isCollapsed),
+        );
+    };
```

Comment:
Here you could use UseCallback to avoid re creating the function on each update

Reply:
This is not necessary since we would also need to wrap the component in memo to prevent this